### ISSUE TITLE
Mod overwriting capes, impossible to have custom capes

### ISSUE THREAD

**Issue by DelofJ** [zlainsama/SkinPort#31]
I use a custom vanilla cape by changing the texture of the cape file which is in the "assets / skins" folder (Minecraft uses the skins and the cape that are already downloaded in "assets / skins" and if it doesn't find them it re-downloads them, this allows to manipulate them and to have personalized capes)
I noticed that another mod to have the skins in Minecarft 1.7.10, "Lumy Skin Patch" which is less advanced than this one, uses the Minecraft skins by default using the "assets/skins" folder which allows you to have more possibilities in handling
![Lumy Skin Patch - cape](https://user-images.githubusercontent.com/50058956/145728715-b04d3132-2a51-4e99-903e-2b089c0569d0.png)
![SkinPort - cape](https://user-images.githubusercontent.com/50058956/145728706-392bdd12-6c7a-46a8-9ffb-5f3883b0446a.png)
Being more convenient, is it possible that this mod uses the skins that are in "assets/skins" like in Minecraft vanilla or is it possible to modify them in some other way please ?

**Comment by zlainsama** [comment]
The idea of using assets/skins was avoided in early development of my skins mods, simply because the assets are managed by other programs such as the game launcher.
Technically, it is possible to modify the code to use assets/skins, but you have to do it on your own or have someone else do it for you.

**Comment by DelofJ** [comment]
Update : 
I don't know if this problem is related, but I found a bug with capes
If you have an OptiFine cape, the mod overwrites the OptiFine cape with the vanilla cape
There was already an issue [here](https://github.com/zlainsama/SkinPort/issues/5) and [here](https://github.com/zlainsama/SkinPort/issues/24) but the issue keeps happening (I'm using the latest version of both mods and Forge)
If you want screenshots, these are the same as above

**Comment by zlainsama** [comment]
Currently, if SkinPort finds a cape through it's configured sources for a player, it will use that for that player.
There are several ways to avoid this:
1. Disable all sources that provide vanilla capes, set 'useMojang' to 'false' and 'useCrafatar' to 'false'. (Unfortunately, this also disables skins from these sources)
2. Manually put the desired cape inside '.minecraft\cachedImages\capes' folder with a file name formatted like this username.png.
3. Setup a custom web server/proxy that rejects all vanilla capes and use that server as the sole source for the mod.

**Comment by DelofJ** [comment]
Putting a cape file into the capes folder worked, thanks
But it's not the same cape as the OptiFine cape, I needed to fetch a cape file which is not the optifine one (OptiFine as a weird cape system)
And unfortunately, I think a lot of people won't look for a cape file to put in this folder but want their OptiFine cape
I don't know if it's possible to disable overwrite just for the cape

**Comment by zlainsama** [comment]
Yeah, if I remembered correctly, Optifine does auto conversion for their capes.
You can also just disable your vanilla cape on Minecrat.net, this should cause the mod to let the game choose the cape once all the cache got cleared.

**Comment by DelofJ** [comment]
When I disable my vanilla cape, even if in vanilla I don't see my cape, with the mod it keeps showing up
The OptiFine cape appears for a very short time when I go to a 3rd person world but the vanilla cape comes back

**Comment by zlainsama** [comment]
Keep your vanilla cape disabled, wait about an hour or so, check to see if it still shows with the mod.
This is to make sure sources like Crafatar has actually updated their cache about your skin/cape.

**Comment by DelofJ** [comment]
This also worked, thanks, but personally I think I'll keep my cape enabled and put my cape inside the cachedImages folder
